app/model/LTRegexPatternRule.java

```diff
@@ -0,0 +1,169 @@
+/* LanguageTool, a natural language style checker
```

Comment:
Ah interesting, LT is under LGPL. Not a lawyer but I believe copying this code into the project would make Typerighter LGPL too.

I don’t think we currently have a licence in the project? Either way, most of the stuff at the G is Apache.

Not the biggest deal but since we want Typerighter to be usable by other orgs maybe worth doing the reflection trick instead?

Happy to catch up in person to explain further, software licensing is a mess and I am far from an expert on it or what the general approach is at the Guardian

Reply:
@mbarton that's a really good catch! TIL: 'For proprietary software, code under the LGPL is usually used in the form of a shared library, so that there is a clear separation between the proprietary and LGPL components.' Let's catch up.

Reply:
I agree with this analysis. Copying from LT is best avoided. We could perhaps submit a PR that changes the access on the class if there is a good reason for it to be public.